extensions/wikia/FacebookClient/FacebookClientController.class.php

```diff
@@ -126,17 +126,24 @@ private function base64UrlDecode( $input ) {
 	}
 
 	/**
-	 * Disconnect the user from Facebook
+	 * Disconnect the user from Facebook. This can occur in one of two ways, either when the user
+	 * deletes the Wikia App from facebook, of when they explicitly disconnect via Special:Preferences.
+	 * If it comes from Facebook, the request is internal and is sent by FacebookClientController::deauthorizeCallback.
+	 * If it comes explicitly from the user, the request is external and is sent by preferences.js::disconnect.
 	 *
-	 * @requestParam user This is a user object.  Only works for internal calls
+	 * @requestParam user This is a user object.
 	 */
 	public function disconnectFromFB() {
-		$user = $this->getVal( 'user', null );
 
-		if ( $user ) {
-			$user = User::newFromId( $user );
-		} else {
+		if ( $this->request->isInternal() ) {
+			$userId = $this->getVal( 'user' );
+			$user = User::newFromId( $userId );
```

Comment:
Checked in what way? ```deauthorizeCallback``` does a check [here](https://github.com/Wikia/app/blob/dev/extensions/wikia/FacebookClient/FacebookClientController.class.php#L90) which exits if it doesn't get an id before calling this method. The previous version didn't do any additional checking.